### PR TITLE
DROOLS-3425 : Unexpected behavior of docks when multiple editors are opened

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-3425

https://github.com/kiegroup/kie-wb-common/pull/2503
https://github.com/kiegroup/drools-wb/pull/1082
https://github.com/kiegroup/jbpm-designer/pull/837
https://github.com/kiegroup/jbpm-wb/pull/1309
https://github.com/kiegroup/optaplanner-wb/pull/327

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenterTest.java [hunk 1]
```diff
@@ -76,11 +78,15 @@
...
     @Mock
     private SaveAndRenameCommandBuilder<GuidedDecisionTable52, Metadata> saveAndRenameCommandBuilder;
+    @Mock
+    private AuthoringWorkbenchDocks docks;
 
     @Override
     protected GuidedDecisionTableEditorPresenter getPresenter() {
         return new GuidedDecisionTableEditorPresenter(view,
                                                       dtServiceCaller,
+                                                      docks,
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenterTest.java]: Either use `@Mock` or `mock(..)` unless you had good reason?!

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableEditorMenusTest.java [hunk 2]
```diff
@@ -338,6 +343,8 @@ public void setup() {
 
         final GuidedDecisionTableEditorPresenter wrapped = new GuidedDecisionTableEditorPresenter(view,
                                                                                                   dtServiceCaller,
+                                                                                                  docks,
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableEditorMenusTest.java]: Either use `@Mock` or `mock(..)` unless you had good reason?!

**Comment by manstis** [comment]
@gitgabrio I thought you'd want to review given the changes to `scesim`

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenterTest.java (hunk 1 above)
Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenterTest.java]: Fixed and updated the test coverage for this item.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableEditorMenusTest.java (hunk 2 above)
Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableEditorMenusTest.java]: Fixed, the test done for the base covers this unused mock.

File: drools-wb-screens/drools-wb-drl-text-editor/drools-wb-drl-text-editor-client/src/main/java/org/drools/workbench/screens/drltext/client/editor/DRLEditorPresenter.java [hunk 3]
```diff
@@ -193,9 +195,15 @@ protected void save(String commitMessage) {
...
 
+    @Override
+    protected String getEditorIdentifier() {
+        return EDITOR_ID;
+    }
+
     @OnClose
     public void onClose() {
         versionRecordManager.clear();
+        super.onClose();
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-drl-text-editor/drools-wb-drl-text-editor-client/src/main/java/org/drools/workbench/screens/drltext/client/editor/DRLEditorPresenter.java]: I think we could add `@Override` annotation.

Comment (jomarko) [reply drools-wb-screens/drools-wb-drl-text-editor/drools-wb-drl-text-editor-client/src/main/java/org/drools/workbench/screens/drltext/client/editor/DRLEditorPresenter.java]: Ok, please assume the same comment for whole PR, I mean `onClose` method is updated multiple times.

**Comment by Rikkola** [comment]
@jomarko I think I missed this PR when I added the Overrides, adding them in.

**Pull request merged**